src/renderer/atlas/AtlasEngine.cpp

```diff
@@ -750,9 +805,31 @@ void AtlasEngine::_flushBufferLine()
     }
 }
 
+void AtlasEngine::_mapCustomGlyphs(size_t offBeg, size_t offEnd)
+{
+    auto& row = *_p.rows[_api.lastPaintBufferLineCoord.y];
+    auto initialIndicesCount = row.glyphIndices.size();
+    const auto shift = gsl::narrow_cast<u8>(row.lineRendition != LineRendition::SingleWidth);
+    const auto colors = _p.foregroundBitmap.begin() + _p.colorBitmapRowStride * _api.lastPaintBufferLineCoord.y;
+    const auto base = reinterpret_cast<const u16*>(_api.bufferLine.data());
+    const auto len = offEnd - offBeg;
+
+    row.glyphIndices.insert(row.glyphIndices.end(), base + offBeg, base + offEnd);

```

Comment:
I don't totally understand what this means for glyph indices. What are we storing for normal characters? What are we storing for box drawing ones?

Reply:
I'll add a comment. For "custom" glyphs we store the UTF16 code units in the glyph indices array. The reason I do this as opposed to using a 2nd array is so that the hashmap handling code is simpler in `BackendD3D`.

At some point my hope is to move the DWrite handling into the 2 backends. That way there won't be any `row`s and no `glyphIndices` anymore because we'll write the `QuadInstance`s directly into the buffer as we're shaping the text. That way this ugliness will also be gone.